src/weights.jl

```diff
@@ -2,44 +2,189 @@
+aweights(vs::RealVector) = AnalyticWeights(vs)
+aweights(vs::RealArray) = AnalyticWeights(vec(vs))
 
-Construct a `WeightVec` with weight values `vs` and sum of weights `wsum`.
-If omitted, `wsum` is computed.
 """
-function WeightVec{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs))
-    return WeightVec{S, eltype(vs), V}(vs, s)
+    cfactor(w::AnalyticWeights, true)
+
+``\\frac{1}{\sum w - \sum w / \sum {w^2}}``
+"""
+function cfactor(w::AnalyticWeights, ::Type{Val{true}})
+    s = sum(w)
+    sum_sn = 0.0
+    for x in w
+        sum_sn += (x / s) ^ 2
+    end
+
+    1 / (s * (1 - sum_sn))
 end
 
+@weights FrequencyWeights
+
+"""
+    FrequencyWeights(vs, wsum=sum(vs))
+
+Construct a `FrequencyWeights` vector with weight values `vs` and sum of weights `wsum`.
+
+Frequency weights describe the number of cases (or frequency) in which each observation
+was observed. These weight may also be referred to as case weights or repeat weights.
+"""
+FrequencyWeights{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs)) =
+    FrequencyWeights{S, eltype(vs), V}(vs, s)
+
+"""
+    fweights(vs)
+
+Construct a `FrequencyWeights` vector from a given array.
+See the documentation for `FrequencyWeights` for more details.
+"""
+fweights(vs::RealVector) = FrequencyWeights(vs)
+fweights(vs::RealArray) = FrequencyWeights(vec(vs))
+
+"""
+    cfactor(w::FrequencyWeights, true)
+
+``\\frac{1}{\sum{w} - 1}``
 """
-    weights(vs)
+cfactor(w::FrequencyWeights, ::Type{Val{true}}) = 1 / (sum(w) - 1)
 
-Construct a `WeightVec` from a given array.
+@weights ProbabilityWeights
+
+"""
+    ProbabilityWeights(vs, wsum=sum(vs))
+
+Construct a `ProbabilityWeights` vector with weight values `vs` and sum of weights `wsum`.
+
+Probability weights represent the inverse of the sampling probability for each observation,
+providing a correction mechanism for under- or over-sampling certain population groups.
+These weight may also be referred to as sampling weights.
+"""
+ProbabilityWeights{S<:Real, V<:RealVector}(vs::V, s::S=sum(vs)) =
+    ProbabilityWeights{S, eltype(vs), V}(vs, s)
+
+"""
+    pweights(vs)
+
+Construct a `ProbabilityWeights` vector from a given array.
+See the documentation for `ProbabilityWeights` for more details.
 """
-weights(vs::RealVector) = WeightVec(vs)
-weights(vs::RealArray) = WeightVec(vec(vs))
+pweights(vs::RealVector) = ProbabilityWeights(vs)
+pweights(vs::RealArray) = ProbabilityWeights(vec(vs))
 
-eltype(wv::WeightVec) = eltype(wv.values)
-length(wv::WeightVec) = length(wv.values)
-values(wv::WeightVec) = wv.values
-sum(wv::WeightVec) = wv.sum
-isempty(wv::WeightVec) = isempty(wv.values)
+"""
+    cfactor(w::ProbabilityWeights, true)
+
+``\\frac{n}{(n - 1) \sum w}`` where `n = length(w)`
+"""
+function cfactor(w::ProbabilityWeights, ::Type{Val{true}})
+    s = sum(w)
+    n = length(w)
+    return n / (s * (n - 1))
+end
 
-Base.getindex(wv::WeightVec, i) = getindex(wv.values, i)
-Base.size(wv::WeightVec) = size(wv.values)
+"""
+    eweights(n, [λ])
```

Comment:
`λ=0.99`, and drop the brackets. Or maybe drop the default value, as it seems quite arbitrary and unlikely to be exactly what people need?